### PR TITLE
Rewrite alerts management

### PR DESCRIPTION
**Pull request by ajronn**

### CODE REVIEW

File: src/modules/Modules.tsx [hunk 1]
```diff
@@ -10,11 +10,12 @@ import ModulesRouter from './ModulesRouter';
 const Modules = () => {
   return (
     <>
-      <AlertsManager />
       <AuthProvider>
         <PatternsProvider>
           <TechnologiesProvider>
-            <ModulesRouter />
+            <AlertsProvider>
```
Comment (polubis) [inline src/modules/Modules.tsx:16]: What if i would like to display Alert in AuthProvider ? This should be placed higher

File: src/shared/providers/alerts/AlertsProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+const Context = createContext(STATE);
+
+class Provider extends React.Component<AlertsProvider.Props, typeof STATE> {
+    sleep = (milliseconds) => {
+        return new Promise(resolve => setTimeout(resolve, milliseconds))
+    }
+
+    addAlert = async (text: string, type: 'warning' | 'error' | 'success' | 'info') => {
+        let id = 0;
```
Comment (polubis) [inline src/shared/providers/alerts/AlertsProvider.tsx:34]: Please add some white space to make this code prettier. Also format this file - stop using one line ifs

File: src/ui/alert/Alert.tsx [hunk 3]
```diff
@@ -1,50 +1,38 @@
...
-    type?: Types;
-    onClose(): void;
+    message?: string;
+    type?: 'warning' | 'error' | 'success' | 'info';
+    className?: string;
+    time?: number;
+  }
+}
+
+const Alert = ({ message, type = 'error', className = "", time = 5000 }: Alert.Props) => {
```
Comment (polubis) [inline src/ui/alert/Alert.tsx:14]: This should be presentational component only - all logic to - show / hide alerts should be in Provider.

How to implement:

- if error occurs show - alert set delay 
- If next error occurs - show after first one disappear.
- use RxJs operators  for that concatMap, delay, takeUntil and others

File: src/ui/alert/Alert.tsx [hunk 4]
```diff
@@ -1,50 +1,38 @@
...
 namespace Alert {
-  export type Types = 'warning' | 'error' | 'success' | 'info';
-
   export interface Props {
-    message: string;
-    type?: Types;
-    onClose(): void;
+    message?: string;
+    type?: 'warning' | 'error' | 'success' | 'info';
+    className?: string;
```
Comment (polubis) [inline src/ui/alert/Alert.tsx:9]: Missing unit tests for Provider

**Review by polubis — CHANGES_REQUESTED** [review]

File: src/modules/Modules.tsx (hunk 1 above)
Comment (ajronn) [reply src/modules/Modules.tsx:16]: done

File: src/shared/providers/alerts/AlertsProvider.tsx (hunk 2 above)
Comment (ajronn) [reply src/shared/providers/alerts/AlertsProvider.tsx:34]: done

File: src/ui/alert/Alert.tsx (hunk 3 above)
Comment (ajronn) [reply src/ui/alert/Alert.tsx:14]: All alert logic was replaced to AlertProvider, now it works on a recursive basis: function checks if any message is currently displayed, if so, it waits 300ms (cause it is animation time) and checks again, if not, it displays.

File: src/ui/alert/Alert.tsx (hunk 4 above)
Comment (ajronn) [reply src/ui/alert/Alert.tsx:9]: done

**Pull request closed without merging**